src/components/ReviewComponents/submit-utils.ts

```diff
@@ -0,0 +1,98 @@
+import { Dispatch } from 'redux';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
+import { createApplication, createComponent } from '../../utils/create-utils';
+import { FormState } from '../form-context';
+import { ReviewComponentsFormValues } from './types';
+import { transformResources } from './utils';
+
+const createComponents = async (
+  components: ReviewComponentsFormValues['components'],
+  application: string,
+  namespace: string,
+  dryRun?: boolean,
+) => {
+  return Promise.all(
+    Object.values(components).map((component) => {
+      return createComponent(
+        {
+          name: component.name,
+          gitRepo: component.source,
+          replicas: component.replicas && Number(component.replicas),
+          targetPort: component.targetPort && Number(component.targetPort),
+          resources: component.resources && transformResources(component.resources),
+        },
+        application,
+        namespace,
+        dryRun,
+      );
+    }),
+  );
+};
+
+export const createResources = async (
+  formState: FormState,
+  components: ReviewComponentsFormValues['components'],
+  dispatch: Dispatch,
+) => {
+  const shouldCreateApplication = !formState.existingApplication;
+  let appName = formState.existingApplication;
+  try {
+    if (shouldCreateApplication) {
+      const appData = await createApplication(formState.application, formState.namespace, true);
+      appName = appData.metadata.name;
+    }
+    await createComponents(components, appName, formState.namespace, true);
+  } catch (error) {
+    dispatch(
+      addNotification({
+        variant: 'danger',
+        title: 'Resource creaion failed!!',
```

Comment:
Also why is the alert message `Resource creation failed!!` here? User doesnt know we are doing a dryRun before actually creating the resources so for them expectation remains the same in both cases and the alerts should also be the same otherwise it will be confusing for the users. If the application creation fails in dryRun mode then also IMO we should say `Application creation failed!!` so that the user knows what exactly happened and same goes for components creation.

Reply:
I added `resource` instead of `application` because this form can be used to add components to an existing application as well, at which point no new application is created and this message could be confusing.
But I'll change it if you think it's better.

We also have [HAC-560](https://issues.redhat.com/browse/HAC-560) for the other notifications, maybe we can handle this as part of that ticket?

Reply:
Yes I was also trying to say that if the application creation fails in dry run mode it should say `Application creation failed` and if component creation fails during dry run then it should say `Components creation failed`. Similar to what happens when actual creation of both these resources fails respectively.

Reply:
I would suggest handle both the dryRuns separately so that you can show separate alert messages.

Reply:
Updated to use separate alert messages for both dryRuns.